pkg/api/v1/orchestrator/routes/statusValueKV.go

```diff
@@ -0,0 +1,93 @@
+package routes
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/metal-toolbox/conditionorc/internal/status"
+	"github.com/metal-toolbox/rivets/events/registry"
+	"github.com/nats-io/nats.go"
+	"github.com/pkg/errors"
+
+	rctypes "github.com/metal-toolbox/rivets/condition"
+)
+
+type statusValueKV interface {
+	publish(
+		facilityCode string,
+		conditionID uuid.UUID,
+		controllerID registry.ControllerID,
+		conditionKind rctypes.Kind,
+		newSV *rctypes.StatusValue,
+		create,
+		onlyTimestamp bool,
+	) error
+}
+
+// statusValue implements the statusValueKV interface
+type statusValue struct{}
+
+func initStatusValueKV() statusValueKV {
+	return &statusValue{}
+}
+
+func (s *statusValue) publish(
+	facilityCode string,
+	conditionID uuid.UUID,
+	controllerID registry.ControllerID,
+	conditionKind rctypes.Kind,
+	newSV *rctypes.StatusValue,
+	create,
+	onlyTimestamp bool,
+) error {
+	statusKV, err := status.GetConditionKV(conditionKind)
+	if err != nil {
+		return errors.Wrap(errPublishStatus, err.Error())
+	}
+
+	key := rctypes.StatusValueKVKey(facilityCode, conditionID.String())
+	// create
+	if create {
+		newSV.CreatedAt = time.Now()
+		if _, errCreate := statusKV.Create(key, newSV.MustBytes()); errCreate != nil {
+			return errors.Wrap(errPublishStatus, errCreate.Error())
+		}
+
+		return nil
+	}
+
+	// update
+	currEntry, err := statusKV.Get(key)
+	if err != nil && !errors.Is(err, nats.ErrKeyNotFound) {
+		return errors.Wrap(errPublishStatus, err.Error())
+	}
+
+	// only timestamp update takes the current value
+	var publishSV *rctypes.StatusValue
+	if onlyTimestamp {
+		curSV := &rctypes.StatusValue{}
+		if errJSON := json.Unmarshal(currEntry.Value(), &curSV); errJSON != nil {
+			return errors.Wrap(errUnmarshalKey, errJSON.Error())
+		}
+
+		if curSV.WorkerID != controllerID.String() {
```

Comment:
The idea here was to prevent inadvertent/malicious updates from another controller being accepted.
I'd like to verify the the ephermeral controller IDs work fine, for this this check would have to go away.

Reply:
The controllerID is set to the serverID based on recent changes